leases/utils.py

```diff
@@ -219,3 +221,60 @@ def terminate_lease(
         )
 
     return lease
+
+
+def exchange_berth_for_lease(
+    old_lease: BerthLease,
+    new_berth: Berth,
+    switch_date: date,
+    old_lease_comment: str,
+    new_lease_comment: str,
+) -> (BerthLease, BerthLease):
+    # Avoid circular imports
+    from leases.models import BerthLease
+
+    end_date = old_lease.end_date
+    contract = None
+    try:
+        contract = old_lease.contract
+    except old_lease.contract.DoesNotExist:
```

Comment:
Will this work even when `contract` is `None`?

Reply:
There also a test for this ;) `leases.tests.test_lease_mutations.test_switch_berth_without_contract`

Reply:
Although I should set the contract as `None` in that test..

Reply:
Yeah, because if `old_lease.contract` is `None`, it would probably not have the `DoesNotExist` field 🤔

Reply:
If `old_lease.contract` is `None`, and you try to access it `old_lease.contract`, it will throw an error:
```
leases.models.BerthLease.contract.RelatedObjectDoesNotExist: BerthLease has no contract.
```
which can be caught with `old_lease.contract.DoesNotExist`

Reply:
![Uploading image.png…]()

Reply:
Ah, I did not think this quite through. Yea, you are correct, it won't work if it is None.

Probably a better solution would be to replace this block with:
```
    if hasattr(old_lease, "contract"):
        contract = old_lease.contract
```

Reply:
So you can't catch the exception if the variable does not have it :derp:

Reply:
Yep. I was blinded by the passing test that was still working according to the old logic :grin: 

Okay, it's fixed now.